akka-stream/src/main/scala/akka/stream/impl/ActorMaterializerImpl.scala

```diff
@@ -209,6 +214,28 @@ private[akka] class SubFusingActorMaterializerImpl(
       context.asInstanceOf[ActorCell].removeFunctionRef(ref)
     case GetChildren =>
       sender() ! Children(context.children.toSet)
+    case GetChildrenSnapshots =>
+      // Arbitrary timeout but should always be quick, the failure scenario is that
+      // the child/stream stopped, and we do retry
+      implicit val timeout: Timeout = 1.second
+      implicit val scheduler = context.system.scheduler
+      val futureSnapshots: Iterable[Future[Option[StreamSnapshot]]] =
+        context.children.map(
+          child =>
+            retry(
+              { () =>
+                (child ? ActorGraphInterpreter.Snapshot).mapTo[StreamSnapshot].map(s => Some(s)).recover {
+                  case ex: AskTimeoutException
+                      if ex.getMessage
+                        .contains("had already been terminated") || context.child(child.path.name).isEmpty =>
```

Comment:
accessing `context.child` from inside Future `recover`? can we retry all requests instead if one fails?

Reply:
Ups, had bad luck while thinking there... 

Yeah, retry everything from start should work as long as the set of children isn't rapidly changing, then it'l just continue failing.